src/Files/DTO/InlineFile.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Files\DTO;
+
+use WordPress\AiClient\Common\Contracts\WithJsonSchema;
+use WordPress\AiClient\Files\Contracts\FileInterface;
+
+/**
+ * Represents a file with inline base64-encoded data.
+ *
+ * This DTO is used for files that are embedded directly in the request as base64 data,
+ * commonly used for small files or when direct data transfer is preferred.
+ *
+ * @since n.e.x.t
+ */
+class InlineFile implements FileInterface, WithJsonSchema
+{
+    /**
+     * @var string The MIME type of the file.
+     */
+    private string $mimeType;
+
+    /**
+     * @var string The base64-encoded file data.
+     */
+    private string $base64Data;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $mimeType The MIME type of the file.
+     * @param string $base64Data The base64-encoded file data.
+     */
+    public function __construct(string $mimeType, string $base64Data)
+    {
+        $this->mimeType = $mimeType;
+        $this->base64Data = $base64Data;
```

Comment:
We should sanitize the `$base64data` here, to make sure at the root that it's always _only_ the data. Some providers may return base64 data _URLs_, and some developers may pass those around, hence we need to sanitize here to strip the respective prefix if it is one.

For example:
```suggestion
        $this->mimeType = $mimeType;
        $this->base64Data = preg_replace( '/^data:([a-z0-9-]+\/[a-z0-9-]+);base64,/', '', $base64Data );
```

Potentially taking this further, @borkweb and I had discussed in an earlier architecture PR to make the `$mimeType` parameter optional. Ideally the developer would still pass it just to be safe, but if they passed a full data _URL_, that alone would be sufficient, as we would be able to parse out the MIME type. WDYT?

Reply:
Good thoughts! I went about this slightly differently. The most notable difference, and I'm happy to discuss this, is that I'm a bigger fan of validation for this sort of thing than sanitization. Since this is programmatic, an invalid base64 value should not be based; if it is, we want to catch where it's happening rather than push it through. I believe sanitization is based used for raw user input.

From there, the mime type is determined in one of three ways:
1. Uses the `$mimeType` if explicitly set
2. Parses the mime type from the base64 data if present
3. Defaults to `text/plain` per [RFC 2397](https://www.rfc-editor.org/rfc/rfc2397.html) if absent

Tentatively resolved in ac3c9e7

Reply:
I see what you're saying, but in practice the way you implemented it this now leads to the `$base64Data` property to not only contain the base64 data, but also that data URL prefix - in other words, it contains a data URL, not the actual data, which I think is confusing.

I think we should allow both here and sanitize, because there is no need to prohibit either of the two ways. Thinking about it further, the only actual problem is when someone passes _only_ the base64 data while omitting the MIME type.

So how about this?
* If `$base64Data` is provided without data URL prefix _and_ there is no `$mimeType` specified, we have to fail with an exception.
* Otherwise, we can either take the specified `$mimeType` or infer it from the prefixed `$base64Data`.
* In any case, the `$base64Data` should then be sanitized to be purely that data.

We could add another "extra" method to the class like `getDataUrl()` which returns the data URL, i.e. the combination of MIME type and base64 data.

WDYT?

Reply:
@felixarntz Ah, I see what you're saying. Somehow I got it in my head that there would always be the "data:" suffix. I can fix that.

Question, if they _do_ provide the `data:` suffix, using the RFC 2397 spec, should we fall back on `text/plain` as the mime type if it is omitted from the string — per the specification's instruction to do so? Or throw an exception if it's omitted and `$mimeType` is null?

Reply:
Sorry, do you mean if they _don't_ provide a data URL? In that case they must provide `$mimeType`, otherwise we don't know, so will need to throw an exception. It's probably not a reasonable fallback to just assume `text/plain`.

Reply:
Tentatively resolved in af53eaf

Reply:
@felixarntz This is what I mean:
```mermaid
flowchart TD
      Start([Input: base64Data, mimeType])

      Start --> IsDataUri{Is data URI format?}

      IsDataUri -->|Yes| ExtractFromUri[Extract base64 and<br/>MIME from URI]
      ExtractFromUri --> HasProvidedMime1{Was MIME type<br/>provided?}

      IsDataUri -->|No| IsPlainBase64{Is plain base64?}

      IsPlainBase64 -->|No| InvalidFormat[Error: Invalid format]

      IsPlainBase64 -->|Yes| HasProvidedMime2{Was MIME type<br/>provided?}
      HasProvidedMime2 -->|No| MimeRequired[Error: MIME type required<br/>for plain base64]
      HasProvidedMime2 -->|Yes| UseProvided2[Use provided MIME type]

      HasProvidedMime1 -->|Yes| UseProvided1[Use provided MIME type]
      HasProvidedMime1 -->|No| HasExtracted{Does URI contain<br/>MIME type?}

      HasExtracted -->|Yes| UseExtracted[Use extracted MIME type]
      HasExtracted -->|No| UseDefault[Use text/plain]

      style Start fill:#90EE90
      style InvalidFormat fill:#FFB6C1
      style MimeRequired fill:#FFB6C1
```

The focus here being that a Data URI defaults to `plain/text` per the spec. So, technically, a plain base64 value **needs** a specified MIME type, whereas a Data URI does not. But that's per the spec. If we want a Data URI to require an explicit MIME type that's fine — that's the question.

Reply:
From call with @felixarntz:

We are going to have the Data URI require explicit MIME type declaration, either through the parameter or inside of the Data URI itself.